test/Microsoft.VisualStudio.Composition.Analyzers.Tests/VSMEF003ExportTypeMismatchAnalyzerTests.cs

```diff
@@ -421,4 +421,34 @@ class TestClass
 
         await VerifyCS.VerifyAnalyzerAsync(test);
     }
+
+    [Fact]
+    public async Task ExportingWrongUnboundGenericWithNonGenericInterfaces_ForcesEvaluationOfAllInterfaces()
+    {
+        // Reproduces a bug seen related to unbound generic types.
+        string test = """
+            using System;
+            using System.Composition;
+
+            interface IValue<T>
+            {
+            }
+
+            interface IOther<T>
+            {
+            }
+
+            interface IFirstInterface
+            {
+            }
+
+            [Export(typeof(IOther<>))]
```

Comment:
This is an interesting case. Do you have tests for an open generic export like this being imported _multiple times_ with distinct generic type arguments? I wonder whether your contract name collision test will mis-fire and say that a MEF part is importing the same contract twice when it imports `IOther<string>` and `IOther<int>`.